spinoso-string/src/enc/utf8/owned.rs

```diff
@@ -146,6 +147,16 @@ impl Utf8String {
         self.try_push_codepoint(int)
     }
 
+    #[inline]
+    pub fn codepoints(&self) -> Result<Codepoints, CodepointsError> {
+        if let Ok(s) = self.inner.as_slice().to_str() {
```

Comment:
this should use `self.is_valid_encoding()` which is SIMD-optimized.

Reply:
I end up using a single `simdutf8::basic::from_utf8()` in b8729b7.